### PR TITLE
upgrade opentelemetry-proto from v0.11.0 to v0.15.0

### PR DESCRIPTION
**Pull request by esigo**
upgrade opentelemetry-proto to the latest release

## Changes

upgrade opentelemetry-proto to v0.15.0

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: third_party_release [hunk 1]
```diff
@@ -5,6 +5,6 @@ benchmark=v1.5.3
 googletest=release-1.8.0-2523-ga6dfd3ac
 ms-gsl=v1.0.0-393-g6f45293
 nlohmann-json=v3.9.1
-opentelemetry-proto=v0.11.0
+opentelemetry-proto=v0.11.0-16-g3c2915c
```
Comment (ThomsonTan) [inline third_party_release:8]: Should here be 0.15.0 as well?

Comment (esigo) [reply third_party_release:8]: It's strange this is what `git submodule` returns.
There were some breaking changes, I need to fix them for the upgrade.

Comment (lalitb) [reply third_party_release:8]: > There were some breaking changes, I need to fix them for the upgrade.

Is it related to proto file changes for logs sdk?

Comment (esigo) [reply third_party_release:8]: yes, it should be related to log.

**Comment by esigo** [comment]
closing as it is blocked by #1291.

**Pull request closed without merging**